### PR TITLE
feat: split create update delete actions

### PR DESCRIPTION
**Pull request by mxkaske**
Removed the `Dialog` from the form to be able to nest it inside of Dropdown (see [shadcn doc](https://ui.shadcn.com/docs/components/dialog#notes)).

Updated the `MonitorForm` to allow `defaultValues` for the "Update" action. Same behaviour of `Dialog` opening. Also included the "Delete" action with a `AlertDialog`.

Only issue is, the on save behaviour for the "Update" dialog seems to be broken. Need to check.

Otherwise small visual fixes here and there!

No real database action connected to the clicks yet!

### CODE REVIEW

File: apps/web/src/app/app/(dashboard)/[workspaceId]/monitors/_components/action-button.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        </AlertDialogContent>
+      </AlertDialog>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Create Monitor</DialogTitle>
+          <DialogDescription>Create a monitor</DialogDescription>
+        </DialogHeader>
+        <MonitorForm
+          id="monitor-update"
+          onSubmit={onUpdate}
```
Comment (mxkaske) [inline apps/web/src/app/app/(dashboard)/[workspaceId]/monitors/_components/action-button.tsx:130]: `onSubmit` not working as expected.

File: apps/web/src/app/discord/page.tsx
Comment (thibaultleouay) [inline apps/web/src/app/discord/page.tsx:1]: 🚀

**Review by thibaultleouay — APPROVED** [review]
We could improve the type later for the create form otherwise LGTM 🚀

File: apps/web/src/components/forms/montitor-form.tsx [hunk 2]
```diff
@@ -41,143 +28,122 @@ import {
...
-export function MonitorCreateForm() {
-  const [saving, setSaving] = React.useState(false);
-  const [open, setOpen] = React.useState(false);
-  const params = useParams();
-  const router = useRouter();
+type Schema = z.infer<typeof insertMonitorSchema>;
+
+interface Props {
+  id: string;
+  defaultValues?: {
```
Comment (thibaultleouay) [inline apps/web/src/components/forms/montitor-form.tsx]: Why don't we use some z.infer<inserMonitorSchema> ?

**Pull request merged**